File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/03_Advantages/javascript/vehicle.js
// Vehicle interface (simulated)
class Vehicle {
    start() {
        throw new Error("start() must be implemented");
    }
}

class Car extends Vehicle {
    start() {
        console.log("Starting the car");
    }
}

class Bike extends Vehicle {
    start() {
        console.log("Starting the bike");
    }
}

class Truck extends Vehicle {
    start() {
        console.log("Starting the truck");
    }
}

module.exports = { Vehicle, Car, Bike, Truck };

File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/03_Advantages/python/vehicle.py
from abc import ABC, abstractmethod


class Vehicle(ABC):
    @abstractmethod
    def start(self):
        pass


class Car(Vehicle):
    def start(self):
        print("Starting the car")


class Bike(Vehicle):
    def start(self):
        print("Starting the bike")


class Truck(Vehicle):
    def start(self):
        print("Starting the truck")

